### PR TITLE
Run both Windows and Linux builds, side by side

### PR DESCRIPTION
**Pull request by DoomGerbil**
I've done a few things here:

1. I've split out the Linux and Win64 builds into separate steps.
1. I've added a second step to the pipeline, so now one will run Windows, the other will run Linux.  These will [execute in parallel](https://buildkite.com/improbable/unrealgdk-premerge/builds/990), assuming that there are free build agents, so the actual build time now vs just building Windows should generally be the same.
1. I've cleaned up and extended the tracing log data throughout all of the scripts, so now there's some solid [log trace data](https://jaeger-query.production-intinf-eu1.i8e.io/trace/bca1dcb93468425ba2a709e990e7f216).
1. Tweaked the logging a little bit to clean up some extraneous/ignored notation data and so that it folds a bit more cleanly.  See [here](https://buildkite.com/improbable/unrealgdk-premerge/builds/990#6a584b3e-1ff3-48d1-ad96-7a3d250cc8fa) for how the logs all break down now.
1. Note that if you just run the `ci/setup-and-build-gdk.ps1` script as before without passing in a platform, the build will default to Win64.  Calling it like `./ci/setup-and-build-gdk.ps1 -target-platform Linux` should give you Linux instead.

### CODE REVIEW

**Comment by DoomGerbil** [comment]
FYI, I'm OOO for a couple of weeks after today, so if y'all are happy with this, feel free to approve and merge it without me.  😄

File: .buildkite/premerge.steps.yaml [hunk 1]
```diff
@@ -22,7 +22,13 @@ common: &common
 
 steps:
   - label: "build-GDK-:windows:"
-    command: "powershell ./ci/setup-and-build-gdk.ps1"
-    <<: *common # This folds the YAML named anchor into this step. Overrides, if any, should follow, not precede. 
+    command: "powershell ./ci/setup-and-build-gdk.ps1 -target-platform Win64"
```
Comment (m-samiec) [inline .buildkite/premerge.steps.yaml]: Shouldn't `target-platform` be `target_platform`? Or is there some automagic character replacement going on?

Comment (m-samiec) [inline .buildkite/premerge.steps.yaml]:
```suggestion
    command: "powershell ./ci/setup-and-build-gdk.ps1 -target_platform Win64"
```

File: .buildkite/premerge.steps.yaml [hunk 2]
```diff
@@ -22,7 +22,13 @@ common: &common
...
+    <<: *common # This folds the YAML named anchor into this step. Overrides, if any, should follow, not precede.
+    artifact_paths:
+      - "UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
+
+  - label: "build-GDK-:linux:"
+    command: "powershell ./ci/setup-and-build-gdk.ps1 -target-platform Linux"
```
Comment (m-samiec) [inline .buildkite/premerge.steps.yaml]:
```suggestion
    command: "powershell ./ci/setup-and-build-gdk.ps1 -target_platform Linux"
```

File: .buildkite/premerge.steps.yaml [hunk 3]
```diff
@@ -22,7 +22,13 @@ common: &common
...
+    artifact_paths:
+      - "UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
+
+  - label: "build-GDK-:linux:"
+    command: "powershell ./ci/setup-and-build-gdk.ps1 -target_platform Linux"
+    <<: *common # This folds the YAML named anchor into this step. Overrides, if any, should follow, not precede.
     artifact_paths:
-      - "UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
\ No newline at end of file
+      - "UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
```
Comment (claireimprobable) [inline .buildkite/premerge.steps.yaml]: Super minor - looking at the Linux build ran in the last check here https://buildkite.com/improbable/unrealgdk-premerge/builds/991#d362a14f-31ef-4632-9ab8-27482caf9b02 it looks like the artifacts were uploaded under the name of UBT-UE4Editor-Win64-Development.txt - i assume this is due to the fact we're building on Windows but might be worth renaming the logs/changing the structure to make it clear which platform output log comes from where?

Comment (m-samiec) [reply .buildkite/premerge.steps.yaml]: Strangely, this seems resolved in the latest round of builds.

**Pull request merged**